### PR TITLE
Moving environment helpers out of Prelude.fs

### PR DESCRIPTION
**Pull request by ncave**
Rationale for this change is to facilitate easier porting and reuse of code through github references.

List of changes:
- Moving the environment and process helpers out of Prelude.fs into a separate file (no code changes).
- Small change in ResizeArray.fs' map function to avoid missing API on some platforms (i.e. List<>.ConvertAll and System.Converter are not implemented on .NET Core).

@cloudRoutine P.S. I apologize again for not properly attributing this project in my first commit when parts of it were added to the Fable project. I was unsure how to properly attribute a public domain project without touching its files. I have submitted a new PR in Fable that hopefully corrects that.

### CODE REVIEW

File: src/Forge.Core/Prelude.Env.fs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+     printfn ""
+     Console.Write("> ")
+     Console.ReadLine () |> String.stripControls
+
+/// Loads the given text into a XmlDocument
+let XMLDoc text =
+    if String.isNullOrEmpty text then null else
+    let xmlDocument = XmlDocument ()
+    xmlDocument.LoadXml text
+    xmlDocument
```
Comment (cloudRoutine) [inline src/Forge.Core/Prelude.Env.fs]: this should go into xlinq.fs

File: src/Forge.Core/Prelude.Env.fs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    printfn ""
+    Console.Write "> "
+    Console.ReadLine () |> String.stripControls
+
+let promptSelect2 text list =
+     printfn text
+     list |> Array.iter (fun (n, v) -> printfn " - %s (%s)" n v)
+     printfn ""
+     Console.Write("> ")
+     Console.ReadLine () |> String.stripControls
```
Comment (cloudRoutine) [inline src/Forge.Core/Prelude.Env.fs]: I don't think these prompt functions should be in the environment module

**Comment by cloudRoutine** [comment]
don't worry about attribution if you're using github references, its builtin that way :+1: 

change the module name to `Forge.Environment` and the filename to `Environment.fs`

**Comment by ncave** [comment]
@cloudRoutine Re: XMLDoc, it's not being used anywhere so I removed it, but I can add it back and change it to use XDocument.Load if you want me to add it to XLinq.fs?

**Comment by cloudRoutine** [comment]
you were right to take it out, i got it mixed up with xdocument in my head

**Pull request merged**